core/src/main/java/org/elasticsearch/index/query/GeoPolygonQueryBuilder.java

```diff
@@ -19,32 +19,44 @@
 
 package org.elasticsearch.index.query;
 
+import org.apache.lucene.search.Query;
+import org.elasticsearch.Version;
 import org.elasticsearch.common.geo.GeoHashUtils;
 import org.elasticsearch.common.geo.GeoPoint;
+import org.elasticsearch.common.geo.GeoUtils;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
 import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.index.fielddata.IndexGeoPointFieldData;
+import org.elasticsearch.index.mapper.MappedFieldType;
+import org.elasticsearch.index.mapper.geo.GeoPointFieldMapper;
+import org.elasticsearch.index.search.geo.GeoPolygonQuery;
 
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Objects;
 
 public class GeoPolygonQueryBuilder extends AbstractQueryBuilder<GeoPolygonQueryBuilder> {
 
     public static final String NAME = "geo_polygon";
 
-    public static final String POINTS = GeoPolygonQueryParser.POINTS;
+    static final GeoPolygonQueryBuilder PROTOTYPE = new GeoPolygonQueryBuilder(null);
 
-    private final String name;
+    private final String fieldName;
 
-    private final List<GeoPoint> shell = new ArrayList<>();
+    private List<GeoPoint> shell = new ArrayList<>();
```

Comment:
Because of the point(List<GeoPoint>) method below. Although I guess I could called shell.empty() there instead of re-initialising?